### PR TITLE
how-css-works-behind-the-scenes

### PR DESCRIPTION
**Pull request by sardug10**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/how-css-works-behind-the-scenes/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+2. Rather than using !important, use the selector specificity carefully.
+
+3. A selector with 1 ID specificity will always be given priority over the selector with 1000 class specificity. The one with 1 class specificity will be prioritized from the one with even 1000 elements specificity.
+
+4. The **universal selector ‘ \* ‘** has specificity (0, 0, 0, 0).
+
+So this finishes the first step of parsing the CSS, i.e, Cascading.
+
+2. **Processing the real values**
```
Comment (lalith1403) [inline articles/how-css-works-behind-the-scenes/index.md:92]: Check the numbering.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Some minor changes are required. I have made a few. Go through those minor errors. Also, technically this article explains the concepts well! Good job!

**Comment by sardug10** [comment]
Hey, I am still not very comfortable with Github, so I just want to know the meaning of the label you have added i.e, **needs front matter**. Can you tell, what does it means and how can I resolve it? I will be very thankful to you! Also, I think I have resolved the issue of capitalizing the first letter of each sentence and the numbers issue in a single commit. So, if you can confirm that too? Thank you again!

**Comment by lalith1403** [comment]
The labels are for the information of the section managers. Information to be passed to the reader is done via GitHub comments, such as this one.

**Comment by sardug10** [comment]
> The labels are for the information of the section managers. Information to be passed to the author is done via GitHub comments, such as this one.

Okay......I am really sorry about that misunderstanding!

**Comment by hectorkambow** [comment]
plagiarism check done 👍

**Comment by sardug10** [comment]
Hey, Do I need to make some changes to this or is it done from my side? Just making sure because this is my first time. Thank you!!

**Comment by hectorkambow** [comment]
@sardug10 It looks like its ready for a final review (from me) - if anything else is needed I'll be sure to ping you 👍

File: articles/how-css-works-behind-the-scenes/index.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+But, there can be a case **when the ID can also be the same, so we will go to the next precedence until we find a winner**.
+
+In that case, we will go to the very last scene we have while resolving the conflicts, and that is **Source order**.
+
+If the last two scenarios, i.e., **importance, and specificity fails, we can decide the style by comparing the source order**. This means if two blocks have the same importance and specificity, then **the one written last will be applied.**
+
+**Important points to remember:**
+
+1. Do not use **!important** until it should not be given the highest priority.
```
[the remark this answers is not in the document]
Comment (sardug10) [reply articles/how-css-works-behind-the-scenes/index.md]: Done!!

**Comment by lalith1403** [comment]
@hectorkambow the author has made the requested changes in the latest commit. Just pinging on his behalf. @sardug10 once you have made the necessary changes, ping us and let us know by dropping a comment. That way we can proceed further easily.

**Comment by sardug10** [comment]
> @hectorkambow the author has made the requested changes in the latest commit. Just pinging on his behalf. @sardug10 once you have made the necessary changes, ping us and let us know by dropping a comment. That way we can proceed further easily.

@lalith1403 I have made the requested changes that @hectorkambow told me to i.e, removing the double-negative in a sentence and I have also responded to him back quoting **"Done"** after solving that issue. If there are some more changes that I have to do, I will do it.

**Comment by sardug10** [comment]
Hello, A quick question:- When will be the article gets published and will I receive any payout from this as it was mentioned in the contribution guidelines.

**Comment by hectorkambow** [comment]
> Hello, A quick question:- When will be the article gets published and will I receive any payout from this as it was mentioned in the contribution guidelines.

👋  @sardug10  Hello - yes payouts [are covered in our docs here](https://github.com/section-io/engineering-education/blob/master/new_contributors/CONTRIBUTING.md#review-approval--payment-processes) - the article will be published after a final review. (Which I believe this article may be ready for a final review - after that enhancement.)

**Comment by sardug10** [comment]
Hello, @hectorkambow...thank you for reviewing my article and publishing it. Just a last thing, I want to join the **sectionio-community on Slack** but I am unable to register to slack and it is telling me to get an invite from the **workspace administrator at sectionio-community**. Can you help me with this?

**Pull request merged**